it/v1/endpoints/RetrieveItsaStatusControllerISpec.scala

```diff
@@ -0,0 +1,184 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.endpoints
+
+import api.models.domain.TaxYear
+import api.models.errors._
+import api.stubs.{AuditStub, AuthStub, DownstreamStub, MtdIdLookupStub}
+import play.api.http.HeaderNames.ACCEPT
+import play.api.http.Status._
+import play.api.libs.json.{JsValue, Json}
+import play.api.libs.ws.{WSRequest, WSResponse}
+import play.api.test.Helpers.AUTHORIZATION
+import support.IntegrationBaseSpec
+
+class RetrieveItsaStatusControllerISpec extends IntegrationBaseSpec {
+
+  "Calling the 'Retrieve ITSA Status' endpoint" should {
+    "return a 200 status code" when {
+      "any valid request is made" in new Test {
+
+        val ifsQueryParams: Map[String, String] = Map("futureYears" -> futureYears, "history" -> history)
+
+        override def setupStubs(): Unit = {
+          DownstreamStub.onSuccess(DownstreamStub.GET, ifsUri, ifsQueryParams, OK, ifsResponse)
+        }
+
+        val response: WSResponse = await(request.withQueryStringParameters("futureYears" -> futureYears, "history" -> history).get())
+        response.status shouldBe OK
+        response.json shouldBe mtdResponse
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
+    }
+
+    "return error according to spec" when {
+
+      "validation error" when {
+        def validationErrorTest(requestNino: String,
+                                requestTaxYear: String,
+                                requestFutureYears: String,
+                                requestHistory: String,
+                                expectedStatus: Int,
+                                expectedBody: MtdError): Unit = {
+          s"validation fails with ${expectedBody.code} error" in new Test {
+
+            override val nino: String        = requestNino
+            override val mtdTaxYear: String  = requestTaxYear
+            override val futureYears: String = requestFutureYears
+            override val history: String     = requestHistory
+
+            val response: WSResponse = await(request.withQueryStringParameters("futureYears" -> futureYears, "history" -> history).get())
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+            response.header("Content-Type") shouldBe Some("application/json")
+          }
+        }
+
+        val input = Seq(
+          ("AA1123A", "2023-24", "true", "true", BAD_REQUEST, NinoFormatError),
+          ("AA123456A", "20199", "true", "true", BAD_REQUEST, TaxYearFormatError),
+          ("AA123456A", "2023-24", "A", "true", BAD_REQUEST, FutureYearsFormatError),
+          ("AA123456A", "2023-24", "true", "B", BAD_REQUEST, HistoryFormatError)
+        )
+        input.foreach(args => (validationErrorTest _).tupled(args))
+      }
+
+      "ifs service error" when {
```

Comment:
Can all these say "downstream" instead of IFS, as technically the controller (& especially its caller) shouldn't know about backend details